docker/docker-compose.yml

```diff
@@ -0,0 +1,73 @@
+version: "3.8"
+
+# Naming the project allows us to use it when deleting containers so we can target only ones
+# generated by this project
+# https://docs.docker.com/compose/environment-variables/envvars/#compose_project_name
+name: cuttle
+
+# Share volumes between images, see https://stackoverflow.com/a/62873977
+#  - https://docs.docker.com/compose/compose-file/07-volumes/
+#  - https://docs.docker.com/compose/compose-file/compose-file-v3/#extension-fields
+x-app-volume:
+  &app-volume
+    - type: bind
+      source: ../src
+      target: /cuttle/src
+
+services:
+  database:
+    container_name: database
+    image: 'postgres:latest'
+    ports:
+      - 5432:5432
+    environment:
+      - POSTGRES_DB=cuttle
+      - POSTGRES_USER=cuttlesworth
+      - POSTGRES_PASSWORD=p4ssw0rd!
+      # Allow all connections, no password required (not recommended)
+      # - POSTGRES_HOST_AUTH_METHOD=trust
+    volumes:
+      - ./data/pgdata:/var/lib/postgresql/data
+    restart: always
+
+  server:
+    container_name: server
+    # Required to be able to hit http://server:1337 from the browser via the client container
+    hostname: server
+    environment:
+      - CUTTLE_DOCKERIZED=true
+      - DATABASE_URL=postgresql://cuttlesworth:p4ssw0rd!@database:5432/cuttle
+      # Can set this to staging for the postgres database to work in docker but may need to modify
+      # the onlyAllowOrigins setting in `config/env/staging.js` file
+      - NODE_ENV=staging
+      - VITE_API_URL=http://localhost:1337
```

Comment:
Only Cypress needs it for e2e purposes, so I'm removing it from server and client.

Reply:
This was a lie, we need it for the sails socket.io instance url and server:1337 won't work... BUTTTTTT, Vite still needs the server:1337 url for the proxy since that runs on the server side from the client (I know that's wacky, but I'll document it so it's more clear.

Reply:
I need a deeper look to wrap my head around this, but that's not blocking atm

Reply:
It's the vite proxy, it runs via node which is server side to do the proxy operations, thus you can use the internal hostname of the docker container in how they talk to one another. If it's a client (browser) side operation, you need to use the exposed localhost urls.